diagnostic_msgs/msg/Heartbeat.msg

```diff
@@ -0,0 +1,4 @@
+# This message can be for example sent along with a large message (like point cloud)
+# to allow another node to measure publishing frequency and delay without subscribing
+# to the large message itself.
+Header header
```

Comment:
I don't think that a header is appropriate. The `frame_id` is irrelevant here.

Reply:
Also what is the policy for the timestamp data? It's presumably important to set that on publish?

Reply:
That's the thing. If you put just stamp there, you have to write all the analytics tools from scratch. If you put there Header, you can rostopic hz, you can add TimeStampStatus diagnostics etc. right away.

And I wouldn't say frame_id is irrelevant. Consider e.g. a topic where you mix messages from multiple sensors and differentiate between them via frame_id (not saying this is optimal, but it is a supported case IMO). Yes, rostopic hz would lose sense on such a topic, but it would still be super-easy to write some kind of demux node which would then provide several streams of diagnostic data, one for each frame_id.

Yes, maybe it should be explicitly specified that stamp should be just whatever stamp was put in the parallel data stream (if there is any), or the publish time (if used "standalone").